kolibri/core/assets/src/views/k-modal.vue

```diff
@@ -0,0 +1,312 @@
+<template>
+
+  <!-- Accessibility properties for the overlay -->
+  <transition name="fade">
+    <div
+      class="modal-overlay"
+      @keyup.esc="emitCancelEvent"
+      @keyup.enter="emitEnterEvent"
+      ref="modal-overlay"
+      id="modal-window"
+    >
+      <div
+        class="modal"
+        ref="modal"
+        :tabindex="0"
+        role="dialog"
+        aria-labelledby="modal-title"
+        :class="size"
+        :style="modalMaxSize"
+      >
+
+        <!-- Modal Title -->
+        <h1
+          class="title"
+          id="modal-title"
+          ref="title"
+        >
+          <!-- Accessible error reporting per @radina -->
+          <span
+            v-if="hasError"
+            class="visuallyhidden"
+          >
+            {{ $tr('errorAlert') }}
+          </span>
+          {{ title }}
+        </h1>
+
+        <form
+          @submit.prevent="emitEnterEvent"
+          @keyup.enter.stop
+        >
+          <!-- Default slot for content -->
+          <div
+            class="content"
+            ref="content"
+            :style="contentSectionMaxHeight"
+          >
+            <slot></slot>
+          </div>
+
+          <div
+            class="actions"
+            ref="actions"
+          >
+            <!-- Slot for buttons -->
+            <slot
+              v-if="$slots.actions"
+              name="actions"
+            >
+            </slot>
+            <template v-else>
+              <k-button
+                v-if="cancelText"
+                name="cancel"
+                :text="cancelText"
+                appearance="flat-button"
+                :disabled="cancelDisabled"
+                @click="emitCancelEvent"
+              />
+              <k-button
+                v-if="submitText"
+                name="submit"
+                :text="submitText"
+                :primary="true"
+                :disabled="submitDisabled"
+                type="submit"
+              />
+            </template>
+          </div>
+        </form>
+      </div>
+    </div>
+  </transition>
+
+</template>
+
+
+<script>
+
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+  import debounce from 'lodash/debounce';
+  import kButton from 'kolibri.coreVue.components.kButton';
+
+  /**
+   * Used to focus attention on a singular action/task
+   */
+  export default {
+    name: 'kModal',
+    components: {
+      kButton,
+    },
+    mixins: [responsiveWindow],
+    $trs: {
+      // error alerts
+      errorAlert: 'Error in',
```

Comment:
this string should be parameterized like `Error in {field}`

(I realize this was a pre-existing issue)